dashboard/app/models/ability.rb

```diff
@@ -201,6 +202,11 @@ def initialize(user)
             )
             can_view_as_user_for_code_review = true
           end
+
+          # Code review V2
+          if user != user_to_assume && !user_to_assume.student_of?(user) && can?(:code_review, user_to_assume)
```

Comment:
What case does `!user_to_assume.student_of?(user)` cover?

Reply:
I copied this from v1, I guess the teacher isn't viewing as a code reviewer like the student is, the teacher is viewing through the teacher view? We should probably run through all of these scenarios in a bug bash and make sure it all works as expected

Reply:
Hmmm, we don't have to hold up this PR, but we should clarify with mike what this pane looks like for the teacher of a student.  Does it looks just like when a (student) peer views it?  If not, what's different?

What happens if a teacher creates a code review?  

Also, I wonder if teachers practice this in PL and what view (student or teacher) they expect to see.

Reply:
That seems right, that we'd want the teacher to be able to participate as a student. We'll want to make sure that scenario works. During the bug bash last week it was decided that the teacher should be able to comment on student's code reviews and delete comments, but there wasn't more clarity given on specific teacher scenario teachers/experience.